adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImpl.java

```diff
@@ -81,43 +84,36 @@ public Integer convertHHMMSSToIntFromNoonOfDayOfService(final String timeAsStrin
     /**
      * This method converts a number of seconds elapsed since noon of say of service to a string formatted as HH:MM:SS.
      *
-     * @param elapsedDurationSinceNoon the number of seconds elapsed since noon of say of service to be converted as a
-     *                                 string formatted as HH:MM:SS.
+     * @param elapsedDurationSinceNoonInSeconds the number of seconds elapsed since noon of day of service to be
+     *                                          converted as a string formatted as HH:MM:SS.
      * @return the human readable string representation of the number of seconds elapsed since noon of day of service.
      * This string is formatted as follows: HH:MM:SS.
      */
-    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoon) {
-        if (elapsedDurationSinceNoon != null) {
-            // Determine number of hours elapsed
-            int hourValue = (int) (TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon) + 12);
-            int minuteValue;
-            int secondValue;
-            // elapsedDurationSinceNoon > 0 for times after noon, e.g 14:00PM is represented a +2*3600s=+7200s
-            // elapsedDurationSinceNoon < 0 for times before noon, e.g 11:00AM is represented as -1*3600s=-3600s
-            if (elapsedDurationSinceNoon > 0) {
-                // Determine number of minutes and seconds elapsed
-                minuteValue = (int) (TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon) -
-                        TimeUnit.HOURS.toMinutes(TimeUnit.SECONDS.toHours(elapsedDurationSinceNoon)));
-                secondValue = (int) (TimeUnit.SECONDS.toSeconds(elapsedDurationSinceNoon) -
-                        TimeUnit.MINUTES.toSeconds(TimeUnit.SECONDS.toMinutes(elapsedDurationSinceNoon)));
+    public String convertIntegerToHMMSS(final Integer elapsedDurationSinceNoonInSeconds) {
+        if (elapsedDurationSinceNoonInSeconds != null) {
+            final LocalTime noon = LocalTime.NOON;
+            // elapsedDurationSinceNoonInSeconds > 0 for times after noon, e.g 14:00PM is represented a +2*3600s=+7200s
+            // elapsedDurationSinceNoonInSeconds < 0 for times before noon, e.g 11:00AM is represented as -1*3600s=-3600s
+            final boolean isTimeGreaterThan24hours =
+                    elapsedDurationSinceNoonInSeconds >= TimeUnit.HOURS.toSeconds(12);
+
+            if (!isTimeGreaterThan24hours) {
+                return String.format("%02d:%02d:%02d", noon.plusSeconds(elapsedDurationSinceNoonInSeconds).getHour(),
```

Comment:
Just curious - did `return noon.plusSeconds(elapsedDurationSinceNoon).toString();` not work here?

Reply:
The String returned by `toString` depends on whether the time has seconds. If the time has seconds we end up with a String formatted as HHMMSS, but if it does not we end up with a String formatted as HHMM. As we want to maintain the HHMMSS format in any case, I had to make this change.

Reply:
Ah, thanks, makes sense 👍